api/v1alpha1/envoyproxy_types.go

```diff
@@ -136,14 +136,15 @@ type EnvoyProxySpec struct {
 	// These settings are applied on backends for which TLS policies are specified.
 	// +optional
 	BackendTLS *BackendTLSConfig `json:"backendTLS,omitempty"`
-
 	// IPFamily specifies the IP family for the EnvoyProxy fleet.
-	// This setting only affects the Gateway listener port and does not impact
-	// other aspects of the Envoy proxy configuration.
-	// If not specified, the system will operate as follows:
-	// - It defaults to IPv4 only.
-	// - IPv6 and dual-stack environments are not supported in this default configuration.
-	// Note: To enable IPv6 or dual-stack functionality, explicit configuration is required.
+	// This setting affects the DNS resolver for the EnvoyProxy fleet.
```

Comment:
this is incorrect, this field is exclusive for the listener / envoy proxy and the cluster IPFamily is derived from the Service / EndpointSlice of the backenRef
cc @juwon8891

Reply:
But right now, from logic the main affect is dnsLookupFamily, or I may missed something.

Reply:
yeah its using the listener IPFamily instead of the addressType field or ipFamily field of the destination, which looks like a bug

Reply:
default < envoyproxy < destination?

Reply:
envoyProxy resource IPFamily is only meant for listener and Envoy Proxy fleet Service spec

BackendRef IPFamily determines cluster setting

Reply:
as we use `::` and `IPv4Compact=true` this knob seems meaningless?

Reply:
I reverted the API doc change, seems this PR didn't related to it.

Reply:
Yeah now it will only be used in the infra manager to create the Service spec